commit 29d779c7029f89e530994c45ae065707ffd62c43
Author: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date:   Thu Mar 6 18:57:40 2014 -0300

    smbreadline: switch to new-style readline typedef
    
    Function, CPFunction, CPPFunction and VFunction typedefs are considered
    old-style (deprecated) starting from readline 4.2 (circa 2001).
    Compatibility typedefs have been in place up to readline 6.2 but were
    removed with the 6.3 release thus causing builds to break.
    Switch to the new-style specific prototyped typedef.
    
    Return value is unused so the callback should still be void (see
    readline/input.c around line 456 in version 6.3).
    
    Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
    Reviewed-by: Jeremy Allison <jra@samba.org>
    Reviewed-by: Andrew Bartlett <abartlet@samba.org>
    
    Autobuild-User(master): Jeremy Allison <jra@samba.org>
    Autobuild-Date(master): Thu Mar 13 00:21:47 CET 2014 on sn-devel-104

diff --git a/libcli/smbreadline/smbreadline.c b/libcli/smbreadline/smbreadline.c
index cff25a7..80e10b0 100644
--- a/libcli/smbreadline/smbreadline.c
+++ b/libcli/smbreadline/smbreadline.c
@@ -141,7 +141,7 @@ char *smb_readline(const char *prompt, void (*callback)(void),
 
 #if HAVE_DECL_RL_EVENT_HOOK
 	if (callback)
-		rl_event_hook = (Function *)callback;
+		rl_event_hook = (rl_hook_func_t *)callback;
 #endif
 	ret = readline(prompt);
 	if (ret && *ret)
